src/main/java/com/java_enterprise/home_work_2/ValueCalculator.java

```diff
@@ -0,0 +1,125 @@
+package com.java_enterprise.home_work_2;
+
+import java.util.Arrays;
+import java.util.Random;
+import java.util.stream.DoubleStream;
+import java.util.stream.IntStream;
+
+public class ValueCalculator {
+    private double[] originArray;
+    private int size;
+    private int halfSize;
+    private int numberOfThreads;
+
+    public ValueCalculator(int size) {
+        this.size = Math.max(size, 1_000_000);
+        this.halfSize = (int) Math.floor(this.size / 2.);
+        this.originArray = new double[this.size];
+        this.numberOfThreads = 1;
+    }
+
+    public void setNumberOfThreads(int numberOfThreads) {
+        this.numberOfThreads = numberOfThreads;
+    }
+
+    public void compute() throws InterruptedException {
+        var startAt = System.currentTimeMillis();
+
+        Arrays.fill(this.originArray, 17.88);
+
+        switch (this.numberOfThreads) {

```

Comment:
What if you have more than 2 threads in the pool?

Reply:
In BRD its not specified))